File: samples/kotlin/src/test/java/cn/staynoob/stuff/java/generic/ConsoleTest.java
package cn.staynoob.stuff.java.generic;

import org.junit.jupiter.api.Test;

import java.util.function.Supplier;

class ConsoleTest {

    private Supplier<String> supplier = () -> "hello world";

    @Test
    void test100() {
        Supplier<String> supplier = () -> "hello world";

//        // doesn't compile
//        Console.debug(supplier);
//        // also doesn't compile
//        Console.debug((Supplier<Object>) supplier);
    }

}

File: samples/kotlin/src/test/kotlin/cn/staynoob/stuff/kotlin/generic/ConsoleTest.kt
package cn.staynoob.stuff.kotlin.generic

import cn.staynoob.stuff.java.generic.Console
import org.junit.jupiter.api.Test
import java.util.function.Supplier

class ConsoleTest {

    @Test
    fun test100() {
        val supplier: Supplier<String> = Supplier {
            "hello world"
        }
        Console.debug(supplier)
    }

}